client/internal/dns/file_parser_linux.go

```diff
@@ -0,0 +1,98 @@
+//go:build !android
+
+package dns
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+const (
+	defaultResolvConfPath = "/etc/resolv.conf"
+)
+
+type resolvConf struct {
+	nameServers   []string
+	searchDomains []string
+	others        []string
+}
+
+func (r *resolvConf) String() string {
+	return fmt.Sprintf("search domains: %v, name servers: %v, others: %s", r.searchDomains, r.nameServers, r.others)
+}
+
+func parseDefaultResolvConf() (*resolvConf, error) {
+	return parseResolvConfFile(defaultResolvConfPath)
+}
+
+func parseBackupResolvConf() (*resolvConf, error) {
+	return parseResolvConfFile(fileDefaultResolvConfBackupLocation)
+}
+
+func parseResolvConfFile(resolvConfFile string) (*resolvConf, error) {
+	file, err := os.Open(resolvConfFile)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	cur, err := os.ReadFile(resolvConfFile)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(cur) == 0 {
+		return nil, fmt.Errorf("file is empty")
+	}
+
+	rconf := &resolvConf{
+		nameServers: make([]string, 0),
```

Comment:
It is true but it is not guaranteed we will append any element. It could be empty nameServers, others, and search domains. In this case, the nameServers will be nil.  What do you think?

Reply:
I think a nil slice is as good as an empty one, but either way it should be consistent. So `searchDomains` should be initialized as well

Reply:
Good point!